### PR TITLE
Combine rolling window function tests

### LINKED ISSUE #7455: [BUG] Multiple definition of `cudf::max_string_sentinel'
The constant array 
```
__constant__ char max_string_sentinel[5]{"\xF7\xBF\xBF\xBF"};
``` 
is defined in a header file (`device_operators.cuh`). Therefore, when that header is included multiple times in multiple, separate compilation units, we will have compiling error saying that there are multiple definitions of `max_string_sentinel`.

### PR DESCRIPTION
**Pull request by mythrocks**
The `tests/` folder has several tests pertaining to `*rolling_window()` functionality:
1. `rolling/`
2. `grouped_rolling/`
3. `lead_lag/`
4. `collect_list/`

This commit consolidates all `rolling_window`-related tests into a single test directory.

Fixes #7455. 
(This peripheral change was required for the consolidated `rolling_test` to link correctly.)

### CODE REVIEW

**Comment by mythrocks** [comment]
The contents of the tests have not been modified at all, except to remove `CUDF_TEST_PROGRAM_MAIN()`.

The only stray modification here is to `include/cudf/utilities/device_operators.cuh`, to force internal linkage for `max_string_sentinel`. @karthikeyann, I wonder if you might be alright with that change.

File: cpp/include/cudf/detail/utilities/device_operators.cuh [hunk 1]
```diff
@@ -92,7 +92,7 @@ struct DeviceCount {
  * values. Also, this char pointer serves as valid device pointer of identity
  * value for minimum operator on string values.
  */
-__constant__ char max_string_sentinel[5]{"\xF7\xBF\xBF\xBF"};
+static __constant__ char max_string_sentinel[5]{"\xF7\xBF\xBF\xBF"};
```
Comment (vuule) [inline cpp/include/cudf/detail/utilities/device_operators.cuh:95]: @kaatish for visibility (wrt https://github.com/rapidsai/cudf/pull/7439/files#r583304683 convo).

**Review by vuule — APPROVED** [review]

File: cpp/include/cudf/detail/utilities/device_operators.cuh (hunk 1 above)
Comment (mythrocks) [reply cpp/include/cudf/detail/utilities/device_operators.cuh:95]: @harrism informs me that he discussed this bit with @karthikeyann, and that they are alright with this part. I have requested Karthik to examine this PR.

**Review by karthikeyann — APPROVED** [review]

**Comment by mythrocks** [comment]
Thank you for the reviews, @karthikeyann, @vuule! I'll merge this now.

**Comment by mythrocks** [comment]
@gpucibot merge

**Comment by mythrocks** [comment]
Ah, shoot. Spotted the mistake on refresh. I'll request a `cmake` review before merging.

**Review by harrism — APPROVED** [review]

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**